source/DIYLinux.rst

```diff
@@ -0,0 +1,153 @@
+and are keen on keeping your Windows installation that you paid $$ or
+$$$$ for. In that case, try it. `There are detailed instructions
+available <https://help.ubuntu.com/community/WindowsDualBoot>`__.
+
+Alternatively, do a native install of Ubuntu, and then virtualize the
+other operating system. Google "windows virtual machine on ubuntu", and
+you will get plenty of step by step tutorials.
+
+Dual boot with Mac OS
+~~~~~~~~~~~~~~~~~~~~~
+
+Don't be crazy! 
+
+Both Linux and MacOS operating systems stem from Unix, an older OS which was fairly heavily copyrighted and licensed. MacOS is based on Darwin (a unix-like operating system), whilst Linux is another Unix-like OS.
+ 
+Practically this means that MacOS and Linux behave similarly and are both much easier to program with than Windows. That said, there are some benefits to Linux for a programmer over MacOS. Firstly you are basically trusted to break and fix most things yourself, whilst you CAN do similar things on Mac, they may be harder to achieve or done in different, more roundabout ways.
+ 
+There are minor inconsistencies between Mac and Linux implementations of various tools, for instance ``grep``. This means that a bash script written for Linux may not work out of the box on Mac. BUt that is a minor issue. 
+
+So the take-home message is, don't bother wasting time with a MacOS - Linux dual boot, unless your life is *really* boring. 
+
+
+Creating partitions during Linux installation
+~~~~
+
+When you install Ubuntu, you will have the opportunity to create separate
+``swap``, ``root`` and ``home`` partitions on your hard disk. ``swap``
+necessarily needs to be a separate partition, while ``root + home`` can be on
+the same partition. 
+
+We suggest that you create a separate ``home`` partition, even though it is not
+necessary — that way, even if you break your linux install, you can easily
+reinstall it by just wiping the root partition, without losing any of your data
+(which sits in ``home``).
+
+If you are unsure about this, just go with the default, or ask one of your instructors/GTAs.
+
+Tweaking your Ubuntu OS
+-----------------------
+
+Once you have installed Ubuntu, there are are many ways in which you can
+tweak your OS environment. For example, see
+http://www.howtogeek.com/tag/ubuntu/ubuntu-tips/.
+
+But be careful, it can be addictive and dangerous to your system's
+stability!
+
+Here are a couple of tweaks to our bash/terminal behavior that are
+recommended.
+
+Opening Nautilus from terminal
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In terminal you can simply enter “f” to open nautilus in current
+directory by doing the following. Firstly make a ``.bash_aliases`` file,
+then open it for editing:
+
+.. code:: bash
+
+    touch ~/.bash_aliases
+    gedit ~/.bash_aliases
+
+Next add to the last line of the file, add:
+
+``alias f='nautilus .'``
+
+Then restart terminal, or in current terminal, type:
+
+.. code:: bash
+
+    source ~/.bash_aliases
```

Comment:
The convention elsewhere in the document (incompletely, I need to proof this) is to precede console lines with `$` as a prompt.